cpp/inc/bond/protocol/simple_json_reader.h

```diff
@@ -179,15 +195,190 @@ class SimpleJsonReader
     friend typename boost::enable_if<is_map_container<X> >::type
     DeserializeMap(X&, BondDataType, const T&, SimpleJsonReader<Buffer>&);
 
-    SimpleJsonReader(const SimpleJsonReader& that, const char* name)
-        : _stream(that._stream),
-          _document(that._document),
-          _value(&(*that._value)[name])
-    {}
+    /// @brief Holds either a buffer and its assoicated stream XOR a pointer
+    /// to some parent StreamHolder.
+    class StreamHolder
+    {
+    public:
+        explicit StreamHolder(typename boost::call_traits<Buffer>::param_type input)
+            : _input(input),
+              _stream(_input.value()),
+              _parent()
+        {
+            BOOST_ASSERT(IsParent());
+            BOOST_ASSERT(!IsChild());
+        }
+
+        explicit StreamHolder(SimpleJsonReader& parent)
+            : _input(),
+              _stream(),
+              // If we're creating a child of a child, use the "root" parent
+              // as our parent instead of having a bunch of intermediate
+              // children as parent.
+              _parent(parent._streamHolder.GetParent())
```

Comment:
Just add `// Resolve the parent` or something like that next to variable. This does not deserve 3 lines of commenting.